tests/e2e/specs/backend/site-editing-templates.test.js

```diff
@@ -0,0 +1,180 @@
+import {
+	activateTheme,
+	canvas,
+	getCurrentSiteEditorContent,
+	insertBlock,
+	trashAllPosts,
+} from '@wordpress/e2e-test-utils';
+import { addQueryArgs } from '@wordpress/url';
+import {
+	getNormalPagePermalink,
+	visitPostOfType,
+} from '@woocommerce/blocks-test-utils';
+import {
+	DEFAULT_TIMEOUT,
+	filterCurrentBlocks,
+	getAllTemplates,
+	goToSiteEditor,
+	saveTemplate,
+	waitForCanvas,
+} from '../../utils';
+
+function blockSelector( id ) {
+	return `[data-type="${ id }"]`;
+}
+
+function defaultTemplateProps( templateTitle ) {
+	return {
+		templateTitle,
+		addedBy: WOOCOMMERCE_ID,
+		hasActions: false,
+	};
+}
+
+function legacyBlockSelector( title ) {
+	return `${ blockSelector(
+		'woocommerce/legacy-template'
+	) }[data-title="${ title }"]`;
+}
+
+const BLOCK_DATA = {
+	'single-product': {
+		attributes: {
+			placeholder: 'single-product',
+			template: 'single-product',
+			title: 'WooCommerce Single Product Block',
+		},
+		name: 'woocommerce/legacy-template',
+	},
+};
+
+const SELECTORS = {
+	blocks: {
+		paragraph: blockSelector( 'core/paragraph' ),
+		singleProduct: legacyBlockSelector(
+			'WooCommerce Single Product Block'
+		),
+	},
+};
```

Comment:
Would it make sense for universal selectors and selector functions above to be part of utils?
We already seem to have some SELECTORS there.

I could imagine this could grow one day to earn its own file :)

Reply:
Had a quick look at  Page Object Pattern, looks really interesting. If you had a link to the discussions on this I'd love to learn more!